pkg/service/active_directory_membership_service.go

```diff
@@ -106,3 +106,68 @@ func (svc *ActiveDirectoryMembershipService) DeleteActiveDirectoryMembership(id
 	}
 	return nil
 }
+
+// RemoveActiveDirectoryMembership - leaves the Active Directory domain.
+//   Required parameters:
+//       id - ID of the active directory.
+//       user - Name of the Activer Directory user with the privilege to leave the domain.
+//       password - Password for the Active Directory user.
+
+//   Optional parameters:
+//       force - Use this option when there is an error when leaving the domain.
+
+func (svc *ActiveDirectoryMembershipService) RemoveActiveDirectoryMembership(id string, user string, password string, force *bool) error {
+
+	if len(id) == 0 || len(user) == 0 || len(password) == 0 {
+		return fmt.Errorf("error: invalid parameter specified id:%v, user:%v, password:%v, force:%v ", id, user, password, force)
+	}
+
+	err := svc.objectSet.Remove(&id, &user, &password, force)
+	return err
+}
+
+// ReportStatusActiveDirectoryMembership - reports the detail status of the Active Directory domain.
+//   Required parameters:
+//       id - ID of the active directory.
+
+func (svc *ActiveDirectoryMembershipService) ReportStatusActiveDirectoryMembership(id string) (*nimbleos.NsADReportStatusReturn, error) {
+
+	if len(id) == 0 {
+		return nil, fmt.Errorf("error: invalid parameter specified id:%v ", id)
```

Comment:
Would it be helpful for the message to include the function name?  Is there a reason to include the word "error"?

Reply:
Yes we can include function name . wrt 'error' we can go either way but including 'error' makes it more explicit.